### DISCUSSION TITLE
How to provide TypeScript source map in publishRelease?

### DISCUSSION THREAD

**Discussion by arkhamvm** [nuxt-community/sentry-module#490] [Q&A]
Hi.
How to provide TypeScript source map in publishRelease?

**Comment by rchl** [comment]
The `publishRelease` option uploads build artifacts to Sentry including source maps so if something doesn't work as expected then you have to be more specific (at least specify your module configuration).

**Comment by arkhamvm** [comment]
@rchl it is export source maps, but JS. I need TS.

```js
['@nuxtjs/sentry', {ignore: ['node_modules']}]
```

```js
	sentry: {
		dsn:          process.env.SENTRY_DSN,
		serverConfig: {
			dsn: process.env.SENTRY_RELAY_DSN,
		},
		lazy:     false,
		disabled: ('production' !== process.env.NODE_ENV),
		tracing:  {
			tracesSampleRate: ('production' === process.env.NODE_ENV ? 0.01 : 1.0),
			vueOptions:       {
				tracing:        true,
				tracingOptions: {
					hooks:           ['mount', 'update'],
					timeout:         2000,
					trackComponents: true,
				},
			},
			browserOptions: {},
		},
		config: {
			environment: process.env.SENTRY_ENV,
		},
		publishRelease: {
			url:        process.env.SENTRY_HOST,
			authToken:  process.env.SENTRY_AUTH,
			org:        process.env.SENTRY_ORG_SLUG,
			project:    process.env.SENTRY_PROJ_SLUG,
			setCommits: {
				auto: true,
			},
		},
	},
```

**Reply by rchl** [reply]
What do you mean by "JS"? Source maps are source maps. They are neither JS nor TS.

What specific TS files are you expecting to have source mappings and what is transforming those TS files to JS? Is there a separate step that does that or is it Nuxt doing the transformation?

Also note that I don't have time to query you about all the details so if you don't provide enough details I'll just ignore your question. Ideally just provide a simple project that reproduces your issue.

**Reply by arkhamvm** [reply]
> What do you mean by "JS"? Source maps are source maps. They are neither JS nor TS.

I mean that in Sentry I see already transpiled JS backtrace instead of TS sources.

> Is there a separate step that does that or is it Nuxt doing the transformation?

I use `@nuxt/typescript-build` (https://typescript.nuxtjs.org/guide/setup)

```
	buildModules: [
		['@nuxt/typescript-build', {
			typescript: {
				workers:     Math.min(8, Math.max(1, os.cpus().length - 2)),
				memoryLimit: 4096,
			},
		}],
	],
```

> Ideally just provide a simple project that reproduces your issue.

I created simple Nuxt TS project, but it requires access to some Sentry DSN in `nuxt.config.ts`:
https://codesandbox.io/s/bold-wave-zbsgif?file=/pages/index.vue

**Reply by rchl** [reply]
This works for me.

I've run `npm run build` and `npm run start` and clicked the error button on the page:

![Screenshot 2023-01-14 at 22 02 32](https://user-images.githubusercontent.com/153197/212496896-ad298dfb-e3c5-4173-ae56-e1c454a5db7c.png)

(Note that if you add `"@nuxtjs/sentry"` to `tsconfig.json` then it will reveal some issues with sentry configuration but it's not relevant to this issue)
